### DISCUSSION TITLE
Is SolidJS still not possible?

### DISCUSSION THREAD

**Discussion by o-az** [mdx-js/mdx#2044] [Q&A] [answered]
I'm asking since I see now that SolidJS has a [jsx-runtime](https://github.com/solidjs/solid/tree/main/packages/solid/h/jsx-runtime) which is what I believe was missing.

My setup:

I'm using `@mdx-js/rollup` with `solid-js` and `vite`. When I add mdx files and try to render I get the following:
```
[vite] Internal server error: Failed to resolve import "react/jsx-runtime" from "src/blog/helloworld.mdx". Does the file exist?
  Plugin: vite:import-analysis
  File: /Users/repo/src/blog/helloworld.mdx
  1  |  /*@jsxRuntime automatic @jsxImportSource react*/
  2  |  import {jsx as _jsx} from "react/jsx-runtime";
     |                             ^
  3  |  function MDXContent(props = {}) {
  4  |    const {wrapper: MDXLayout} = props.components || ({});
```

My Vite setup:

```ts
const config = async (): Promise<UserConfig> => {
  const { default: mdx } = await import('@mdx-js/rollup')

  return {
    plugins: [solidPlugin(), WindiCSS(), tsconfigPaths(), mdx({ remarkPlugins: [] })],
    optimizeDeps: {
      include: ['solid-js/h/jsx-runtime'],
    },
    build: { minify: true, target: 'esnext', polyfillDynamicImport: false },
    /** vitest */
    test: { globals: true },
  }
}

export default defineConfig(config)
```

**Comment by wooorm** [comment]
You are not configuring mdx at all. MDX does now know you are using Solid currently. You need to configure MDX.
You need to choose that new solid JSX runtime in the mdx options.

**Reply by o-az** [reply]
Thank you and thanks to modderme123 on SolidJS Discord, got it working. Current setup:

```ts
const config = async (): Promise<UserConfig> => {
  const { default: mdx } = await import('@mdx-js/rollup')

  return {
    plugins: [
      solid(),
      WindiCSS(),
      tsconfigPaths(),
      mdx({
        remarkPlugins: [],
        jsxImportSource: 'solid-jsx',
        providerImportSource: 'solid-mdx',
      }),
    ],
    server: { host: '0.0.0.0', port: 3000 },
    resolve: {
      alias: {
        '@/': './src',
        '~blog/': './data/blog',
      },
    },
    optimizeDeps: {
      include: ['solid-js/h/jsx-runtime'],
    },
    build: { minify: true, target: 'esnext', polyfillDynamicImport: false },
    /** vitest */
    test: { globals: true },
  }
}

export default defineConfig(config)
```

**Reply by wooorm** [reply]
Are you interested in working (with me) on adding it to the docs? Similar to Svelte I’d say: https://mdxjs.com/docs/getting-started/

**Reply by o-az** [reply]
Yes sir I'm very happy to help. Chat on Discord? Discord id: `Amor Fati#6969`

**Reply by wooorm** [reply]
I don’t have discord, but I was thinking of something like this: https://github.com/mdx-js/mdx/pull/1850

**Reply by o-az** [reply]
sounds good, I'll take care of that.

**Reply by o-az** [reply]
@wooorm you're referring to the setup of vite I have in order to get mdx working with SolidJS? Here's what I'm adding

<details>
    <summary>See screenshots</summary>

![CleanShot 2022-05-18 at 17 50 32](https://user-images.githubusercontent.com/23618431/169168158-278078c4-6534-4b6d-be44-adee6f2924b5.png)
-
![CleanShot 2022-05-18 at 17 51 16](https://user-images.githubusercontent.com/23618431/169168166-05f13bbe-40a6-4287-b1a4-5850a5768f74.png)
-
![CleanShot 2022-05-18 at 17 47 51](https://user-images.githubusercontent.com/23618431/169167863-f77e6851-37c9-4bf3-8970-a053bf614054.png)
-
![CleanShot 2022-05-18 at 17 48 09](https://user-images.githubusercontent.com/23618431/169167873-d5ab256b-2bad-4c1d-80f4-432bffacdf93.png)
</details>

**Reply by wooorm** [reply]
Yes, I’m referring to documentation in Getting started for how to configure MDX with Solid.js.
It’s easier if you create a PR, so I can suggest changes.
Have you created PRs before?
Some early notes:
- Place Solid under React and Preact, next to Svelte, in the JSX section
- I don’t think that Rollup and Vite are required for Solid. Look at the Svelte docs for an example of how to document something without forcing a single integration
- `'solid-mdx'` is potentially (hopefully?) not required for Solid either. The provider is an optional context-based API that MDX can use, but does not require
- Please use JavaScript syntax for code examples, and look at other code examples on how to provide a syntax and a filename associated with an example

**Comment by o-az** [comment, marked as answer]
Thanks to modderme123 on SolidJS Discord and @wooorm, got it working. Current setup:

```ts
const config = async (): Promise<UserConfig> => {
  const { default: mdx } = await import('@mdx-js/rollup')

  return {
    plugins: [
      solid(),
      WindiCSS(),
      tsconfigPaths(),
      mdx({
        remarkPlugins: [],
        jsxImportSource: 'solid-jsx',
        providerImportSource: 'solid-mdx',
      }),
    ],
    server: { host: '0.0.0.0', port: 3000 },
    resolve: {
      alias: {
        '@/': './src',
        '~blog/': './data/blog',
      },
    },
    optimizeDeps: {
      include: ['solid-js/h/jsx-runtime'],
    },
    build: { minify: true, target: 'esnext', polyfillDynamicImport: false },
    /** vitest */
    test: { globals: true },
  }
}

export default defineConfig(config)
```
